pkg/retention/retention.go

```diff
@@ -194,7 +194,10 @@ func GetRetentionTimeMs(retentionHours int, currTime time.Time) uint64 {
 func deleteSegmentsFromEmptyPqMetaFiles(segmentsToDelete map[string]*structs.SegMeta) {
 	for _, segmetaEntry := range segmentsToDelete {
 		for pqid := range segmetaEntry.AllPQIDs {
-			pqsmeta.DeleteSegmentFromPqid(pqid, segmetaEntry.SegmentKey)
+			updatedEmptyPQS := pqsmeta.GetUpdatedPQSMapAfterSegmentRemoval(pqid, segmetaEntry.SegmentKey)
+			if updatedEmptyPQS != nil {
```

Comment:
This function removes the PQMR files, if the `len(updatedEmptyPQS) == 0`. Which also deletes the `pqid.meta` file. 

Is this fine? As this also a kind of write?